### DISCUSSION TITLE
Access QuickFilter from Grid to other Dialog Grid

### DISCUSSION THREAD

**Discussion by axelcasauran** [serenity-is/Serenity#6326] [Help]
Hi,

Is it possible to access and modify QuickFilter from Grid to other Dialog Grid?
I would like to put the account id from grid to the quick filter of the dialog grid...
![2022-03-17 22_43_28-Window](https://user-images.githubusercontent.com/13443821/158830308-ad7a7e79-ebb7-4fb6-91f2-67d556dd1d03.png)

Here's some code snippet that I tried but no luck.
![2022-03-17 22_49_22-Window](https://user-images.githubusercontent.com/13443821/158830926-9a57fe73-3f05-4c60-a809-24440267e276.png)

**Comment by minhhungit** [comment]
~@axelcasauran this demo will help you https://serenity.is/demo/BasicSamples/InitialValuesForQuickFilters~
I did not read carefully, your case is quickfilter in grid editor, I have not tried but maybe above link could help

**Reply by minhhungit** [reply]
ah no, your grid is normal grid, not grid editor :) you bring normal grid to dialog form, this case is not standard

**Comment by minhhungit** [comment]
You maybe need to store your quickfilter into a variable, then add a get/set property in your grid (this property will call quickfiller that you stored to set value to) and set value from main grid (General Ledger)

**GL Account grid:**
```javascript
private yourFilter: Serenity.QuickFilter<Serenity.Widget<any>, any>;

protected createQuickFilters(): void {
    // let base class to create quick filters first
    super.createQuickFilters();   
    
    this.yourFilter = this.findQuickFilter(Serenity.LookupEditor, fld.ShipVia);
}

// define a get/set property

private _quickFilterValue: string;

get quickFilterValue() {
    return this._quickFilterValue;
}

set quickFilterValue(value: string) {
    if (this._quickFilterValue!= value) {
        this._quickFilterValue= value;
        this.yourFilter.values = value;
    }
}
```

**General Ledger GRID:**
```javascript
this.lgGrid = new ....

// set quickfilter value here
this.lgGrid.quickFilterValue = some value
```

This is just pseudo code, I've not tested it yet so I'm not sure 100% if it work

**Reply by axelcasauran** [reply]
it did not work... hmmm i changed my approached to be like this https://github.com/serenity-is/Serenity/discussions/5756 still issue as well..

**Reply by minhhungit** [reply]
I tried, **it works** 

```javascript 
    protected getQuickFilters(): Serenity.QuickFilter<Serenity.Widget<any>, any>[] {
        let filters = super.getQuickFilters();
        filters.push({
            type: Serenity.IntegerEditor,
            field: fld.Fields.NumberField,
            title: 'Number Field',
            handler: h => {
                if (h.active) {
                    (h.request as Serenity.ListRequest).Criteria = Serenity.Criteria.and(h.request.Criteria, [[fld.NumberField], '=', h.value]);
                }
            }
    });

    return filters;
}
        
    private yourFilter: Serenity.IntegerEditor;

    protected createQuickFilters() {
        super.createQuickFilters();

        this.yourFilter = this.findQuickFilter(Serenity.IntegerEditor, fld.NumberField);
    }

    private _quickFilterValue: number;

    get quickFilterValue() {
        return this._quickFilterValue;
    }

    set quickFilterValue(value: number) {
        if (this._quickFilterValue != value) {
        this._quickFilterValue = value;
        this.yourFilter.value = value;
    }
}
```

and call in somewhere, for example in GeneralLedger dialog constructor : 

```javascript 
this.form.YOUR_GRID_FIELD.quickFilterValue = 555;
```
